extensions/email/src/main/java/org/mule/extension/email/api/sender/SenderConnection.java

```diff
@@ -0,0 +1,115 @@
+     * @param connectionTimeout the socket connection timeout
+     * @param readTimeout       the socket read timeout
+     * @param writeTimeout      the socket write timeout
+     * @param properties        the custom properties added to configure the session.
+     */
+    public SenderConnection(EmailProtocol protocol,
+                            String username,
+                            String password,
+                            String host,
+                            String port,
+                            long connectionTimeout,
+                            long readTimeout,
+                            long writeTimeout,
+                            Map<String, String> properties) throws EmailConnectionException
+    {
+        super(protocol, username, password, host, port, connectionTimeout, readTimeout, writeTimeout, properties);
+    }
+
+    /**
+     * Creates a new instance of the connection secured by TLS.
+     *
+     * @param protocol          the protocol used to send mails. Smtp or smtps
+     * @param username          the username to establish connection with the mail server.
+     * @param password          the password corresponding to the {@code username}
+     * @param host              the host name of the mail server.
+     * @param port              the port number of the mail server.
+     * @param connectionTimeout the socket connection timeout
+     * @param readTimeout       the socket read timeout
+     * @param writeTimeout      the socket write timeout
+     * @param properties        the custom properties added to configure the session.
+     * @param tlsContextFactory the tls context factory for creating the context to secure the connection
+     */
+    public SenderConnection(EmailProtocol protocol,
+                            String username,
+                            String password,
+                            String host,
+                            String port,
+                            long connectionTimeout,
+                            long readTimeout,
+                            long writeTimeout,
+                            Map<String, String> properties,
+                            TlsContextFactory tlsContextFactory) throws EmailConnectionException
+    {
+        super(protocol, username, password, host, port, connectionTimeout, readTimeout, writeTimeout, properties, tlsContextFactory);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void disconnect()
+    {
+        // No implementation
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * if the {@link SenderConnection} instance exists, then the validation will always be successful
+     */
+    @Override
+    public ConnectionValidationResult validate()
+    {
+        return success();
+    }
+
+    public Transport getTransport()
+    {
+        try
+        {
+            return session.getTransport(protocol.getName());
```

Comment:
it closes after sending the message, this #getTransport method should be deleted, is not used anymore. Will fix that.